### DISCUSSION TITLE
Wiz ESP32-C3-MINI-1 Invalid Error on v14.3.0

### DISCUSSION THREAD

**Discussion by fcol95** [arendst/Tasmota#22588] [Support request]
Hi!
First time trying to flash Tasmota on a device here!
Just opened a Wiz Smart Plug Type B device, soldered jumper wires on 3.3V/GND/TX/RX/IO8/IO9.
Its got a "Wiz ESP32-C3-MINI-1" module
Managed to get to boot mode and flash the latest release of tasmota32c3.factory.bin (v14.3.0).
Also, used esptool to verify flash after writing it to make sure everything when down fine (see screenshots below).

![image](https://github.com/user-attachments/assets/23b6709b-8eab-4112-a99b-5d32ec33f20a)

![image](https://github.com/user-attachments/assets/b7dec003-4a4b-484d-8248-859d326f88a9)

However, when I remove the boot pins and power cycle to start the device, I get the following error (invalid header: 0x39baa2fc) and it just boot cycles from then.

![image](https://github.com/user-attachments/assets/d0f9933e-3539-4b48-aa6e-fc948fd96c1b)

First lines at reset that prints:
```
ESP-ROM:esp32c3-api1-20210207
Build:Feb  7 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x9 (SPI_FAST_FLASH_BOOT)
Saved PC:0x4004c0d4
```
(is it normal that the build date doesn't fit with the release date of v14.3.0? tried both the bin on the release webpage and tasmota-firmware git with same result)

Any idea for a fix?
Thanks!

**Comment by fcol95** [comment]
FYI just built latest commit on development branch (ccd76b26ed8c309b3a413b39535d5309bb1abc9e) and did a build for tasmota32c3 environment on VSCode with platform-espressif32 2024.11.31. Same result as above!

**Comment by arendst** [comment]
The build date of Feb 7 2021 is the Espressif BIOS build date. All messages shown are not related to Tasmota.

Considering the fact that esptools finds 5 serial ports, are you sure you are flashing the correct device (connected to com8)

**Reply by fcol95** [reply]
Good to know about the espressif BIOS messages!
Yes, it's the right serial port, esp tool wouldn't detect the ESP32-C3 chip to flash/verify, and I wouldn't see those messages on the Putty terminal if it wasn't correct.
Also i see the RX/TX led on my uart to USB adapter blinking when flashing.

**Comment by Jason2866** [comment]
Try an official C3 factory firmware. If flashing worked the build firmware is faulty.
Or if downloaded the firmware is corrupted during download

**Reply by fcol95** [reply]
You mean this one? https://ota.tasmota.com/tasmota32/release/tasmota32c3.factory.bin
That's what i tried first and it didn't work.
Then tried this one from github with the same error, https://github.com/arendst/Tasmota-firmware/blob/firmware/firmware/tasmota32/tasmota32c3.factory.bin.
Would be surprising that those two would of been corrupted during download from two different sources and not error during download from my Chrome browser?

Also, tried the web flashing tool first, then ESP tool python module, then the GUI flashing tool, all same failed result.
Any other ideas?
Is it not the right binary?
Or some step to do first before erasing/flashing/verifying that binary? (or flag to put in esptool?)
Or the latest release bin file for C3 is corrupted? (worth trying an older release?)

**Comment by joba-1** [comment]
two more things to check
1) power supply during flash can deliver enough power? Probably, since you verified…
2) my shelly c3 could be flashed but did not boot into tasmota while not powered by 230V - maybe your plug has the same behavior?

**Reply by fcol95** [reply]
Mmm is it possible that the flash encryption from the vendor ESP32-C3 module is preventing me from running a custom firmware?
My symptoms on the console seem very similar to this #1 possible failure...
Not sure if you guys see another option other than desoldering and replacing this ESP32-C3 module with a fresh one?
![image](https://github.com/user-attachments/assets/caba6ecc-2026-47d1-842c-875b9657e847)
https://docs.espressif.com/projects/esp-idf/en/v5.3.1/esp32c3/security/flash-encryption.html#possible-failures

**Reply by fcol95** [reply]
The "--force" was hidding that fault from the other flashing attempts I made, but should of been a hint towards this...
Seems from the espefuse utility that the device has been put to "release mode" with flash encryption enable (and that efuse cannot be written anymore).

![image](https://github.com/user-attachments/assets/b2281778-94b9-4648-9f69-efebe0fe3af8)
...
![image](https://github.com/user-attachments/assets/6652722e-d19d-4832-ba43-d412e6fccc79)

**Reply by fcol95** [reply]
If that's the case, might have been better never flashing that plain text tasmota binary?
Just bricked that ESP32-C3 module I think (any way to add a warning on the getting started page on the website or the readme here?)
![image](https://github.com/user-attachments/assets/92e1bf53-6bbe-4564-9c7e-95954329a08c)
https://docs.espressif.com/projects/esp-idf/en/v5.3.1/esp32c3/security/flash-encryption.html#flash-encryption-configuration
